### PR TITLE
Allow connecting by address

### PR DESCRIPTION
**Pull request by akx**
This PR allows one to pass in the BLE address of the printer instead, to avoid having to wait for autodiscovery.

### CODE REVIEW

File: print.py [hunk 1]
```diff
@@ -30,46 +31,49 @@ def parse_args():
                           search for. If not specified, the script will try to       \
                           auto discover the printer based on its advertised BLE      \
                           service UUIDs. Common names are similar to "GT01", "GB02", \
-                          "GB03".')
+                          "GB03".  You can also pass the device BLE address instead.')
```
Comment (rbaron) [inline print.py]: I think we can simplify the help with something like: `"The printer's Bluetooth Low Energy (BLE) address (MAC address on Linux; UUID on macOS) or advertisement name (e.g.: "GT01", "GB02"). If omitted, the the script will try to auto discover the printer based on its advertised BLE services."`

Another small nit is that it's odd that a `--devicename` arg should accept an address. Maybe we can name it something like `--device-filter` instead?

File: catprinter/ble.py [hunk 2]
```diff
@@ -50,8 +55,23 @@ def chunkify(data, chunk_size):
...
 
 
-async def run_ble(data, devicename, autodiscover, logger):
-    address = await scan(devicename, SCAN_TIMEOUT_S, autodiscover, logger)
+async def get_device_address(devicename: Optional[str]):
+    if devicename:
+        # If devicename is set and smells like an UUID,
+        # just return it.
+        try:
+            return str(uuid.UUID(devicename))
```
Comment (rbaron) [inline catprinter/ble.py]: I think this won't work on Linux, where we need to specify the address in the usual MAC format `AA:BB:CC:DD:EE`.

**Comment by rbaron** [comment]
Thanks again @akx. It looks great modulo two small comments. Can you also update the README.md with the new output of `--help`?

**Comment by akx** [comment]
@rbaron Hi! What two small comments? I didn't see a review or anything..? :)

**Comment by rbaron** [comment]
Oops, I forgot to hit submit review 😬

File: catprinter/ble.py (hunk 2 above)
Comment (akx) [reply catprinter/ble.py]: Ah, right. Only tried on my MBP :) 

Added an alternate path that should detect colon-separated MACs :)

File: print.py (hunk 1 above)
Comment (akx) [reply print.py]: I split the difference and made it just `--device` :)

**Comment by rbaron** [comment]
Thanks a lot @akx, this looks great.

**Pull request merged**